multicloud/src/main/java/de/uniulm/omi/cloudiator/sword/multicloud/pricing/aws/AWSPricingSupplier.java

```diff
@@ -0,0 +1,204 @@
+                                pricingDimensions.add(new PricingDimensions(price.getValue(), entry3.getValue().getUnit(), entry3.getValue().getDescription(), entry3.getValue().getBeginRange(), entry3.getValue().getEndRange()));
+                            }
+                            pricingOnDemandTerms.add(new PricingTerms(pricingDimensions, entry.getValue().termAttributes.get("LeaseContractLength"), entry.getValue().termAttributes.get("OfferingClass"), entry.getValue().termAttributes.get("PurchaseOption")));
+                        }
+                    }
+                    currency = Optional.ofNullable(awsProductAndTerms.terms.get("OnDemand").entrySet().iterator().next().getValue().getAwsDimensionRates().entrySet().iterator().next().getValue().entrySet().iterator().next().getValue().getPricePerUnit().entrySet().iterator().next().getKey());
+                }
+
+                // currently only OnDemand offers are being presented
+                /*if (awsProductAndTerms.terms != null && awsProductAndTerms.terms.get("Reserved") != null) {
+                    for (Map.Entry<String, AWSPriceDimensionsRates> entry : awsProductAndTerms.terms.get("Reserved").entrySet()) {
+                        for (Map.Entry<String, AWSDimensionRates> entry2 : entry.getValue().getAwsDimensionRates().entrySet()) {
+                            List<PricingDimensions> pricingDimensions = new ArrayList<>();
+                            for (Map.Entry<String, AWSBaseTerm> entry3 : entry2.getValue().entrySet()) {
+                                Map.Entry<String, BigDecimal> price = entry3.getValue().getPricePerUnit().entrySet().iterator().next();
+                                pricingDimensions.add(new PricingDimensions(price.getValue(), entry3.getValue().getUnit(), entry3.getValue().getDescription(), entry3.getValue().getBeginRange(), entry3.getValue().getEndRange()));
+                            }
+                            pricingReservedTerms.add(new PricingTerms(pricingDimensions, entry.getValue().termAttributes.get("LeaseContractLength"), entry.getValue().termAttributes.get("OfferingClass"), entry.getValue().termAttributes.get("PurchaseOption")));
+                        }
+                    }
+                    if(!currency.isPresent()) {
+                        currency = Optional.ofNullable(awsProductAndTerms.terms.get("Reserved").entrySet().iterator().next().getValue().getAwsDimensionRates().entrySet().iterator().next().getValue().entrySet().iterator().next().getValue().getPricePerUnit().entrySet().iterator().next().getKey());
+                    }
+                }*/
+
+                Pricing pricing = PricingBuilder.newBuilder()
+                        .id(awsProductAndTerms.product.sku)
+                        .providerId(awsProductAndTerms.product.sku)
+                        .name(awsProductAndTerms.product.attributes.get("usagetype"))
+                        .location(null)
```

Comment:
Probably not needed - null is default value.

Reply:
this method expects `Location location` which I'm not passing in, so null is required